pkg/registry/generic/rest/streamer.go

```diff
@@ -56,6 +59,16 @@ func (s *LocationStreamer) InputStream(apiVersion, acceptHeader string) (stream
 	if err != nil {
 		return
 	}
+
+	if resp.StatusCode == http.StatusInternalServerError {
+		bodyBytes, err := ioutil.ReadAll(resp.Body)
```

Comment:
Oh, I did not realize that. *reads code* In that case I think it's a big mistake to do any interpretation here at all, the stream should be passed to clients exactly as its received.

... why do we even have this object at all? It should just be a wrapper around the rest of our proxying code.

@feihujiang I guess your goal here was to get the HTTP status code repeated back to clients properly? I think you're going to have to fix the interface & callers, there's no way to do that properly with this interface.